File: Java Fundamentals Coding Blocks/13. Recursion/nth_fibonacci.java
// print the nth fibonacci

import java.util.Scanner;

class nth_fibonacci
{
    public static void main(String args[])
    {
        // input number
        System.out.println("Enter the value of n:");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
         System.out.println("\n");
         
        System.out.println(nth_fibonacci(n));
    }

    public static int nth_fibonacci(int n)
    {
       int first = 0, second = 1, next = 0;

       if(n==0 || n==1)
        return n;

       return nth_fibonacci(n-2) + nth_fibonacci(n-1);
    }
}

File: Coding Blocks C++ fundamentals/7 Functions/nth_fibonacci.cpp
// print the nth fibonacci number

#include<iostream>
using namespace std;

int fibonacci(int n)
{
	int a = 0, b = 1, c;

	for (int i = 1; i <= n - 1; i++)
	{
		c = a + b;
		a = b;
		b = c;
	}
	return c;
}


int main()
{
    int n;

    cin >> n;
    cout << fibonacci(n);

    return 0;
}
